### PR TITLE
moved some errors and err checks to core

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: core/trieErr.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	return e.dbIdentifier
+}
+
+// IsInterfaceNil returns true if there is no value under the interface
+func (e *GetNodeFromDBErrWithKey) IsInterfaceNil() bool {
+	return e == nil
+}
+
+// GetNodeFromDBErrorString represents the string which is returned when a getting node from DB returns an error
+const GetNodeFromDBErrorString = "getNodeFromDB error"
```
Comment (iulianpascalau) [inline core/trieErr.go:52]: please move const on the top of the file

File: core/trieErr.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
+package core
```
Comment (iulianpascalau) [inline core/trieErr.go:1]: please rename this file to getNodeFromDBErrWithKey.go

File: core/trieErr.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+// IsInterfaceNil returns true if there is no value under the interface
+func (e *GetNodeFromDBErrWithKey) IsInterfaceNil() bool {
+	return e == nil
+}
+
+// GetNodeFromDBErrorString represents the string which is returned when a getting node from DB returns an error
+const GetNodeFromDBErrorString = "getNodeFromDB error"
+
+// IsGetNodeFromDBError returns true if the provided error is of type getNodeFromDB
```
Comment (iulianpascalau) [inline core/trieErr.go:54]: L54-L95 should be moved in a new file, might call them errorHandling.go

File: core/trieErr.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+func IsGetNodeFromDBError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if IsClosingError(err) {
+		return false
+	}
+
+	if strings.Contains(err.Error(), GetNodeFromDBErrorString) {
```
Comment (iulianpascalau) [inline core/trieErr.go:64]: return directly?

File: core/trieErr.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	return false
+}
+
+// IsClosingError returns true if the provided error is used whenever the node is in the closing process
+func IsClosingError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return strings.Contains(err.Error(), ErrDBIsClosed.Error()) ||
```
Comment (iulianpascalau) [inline core/trieErr.go:77]: extract err.Error() in a variable and use it in those 2 strings.Contains functions?

File: core/trieErr.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// GetNodeFromDBErrWithKey defines a custom error for trie get node
+type GetNodeFromDBErrWithKey struct {
```
Comment (iulianpascalau) [inline core/trieErr.go:11]: missing unit tests for this struct

File: core/trieErr.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply core/trieErr.go:52]: Done.

File: core/trieErr.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply core/trieErr.go:1]: Done.

File: core/trieErr.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply core/trieErr.go:54]: Done.

File: core/trieErr.go (hunk 4 above)
Comment (BeniaminDrasovean) [reply core/trieErr.go:64]: Yes, refactored.

File: core/trieErr.go (hunk 6 above)
Comment (BeniaminDrasovean) [reply core/trieErr.go]: Added.

File: core/getNodeFromDBErrWithKey.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"encoding/hex"
+	"fmt"
+)
+
+// GetNodeFromDBErrorString represents the string which is returned when a getting node from DB returns an error
+const GetNodeFromDBErrorString = "getNodeFromDB error"
+
+// GetNodeFromDBErrWithKey defines a custom error for trie get node
+type GetNodeFromDBErrWithKey struct {
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply core/getNodeFromDBErrWithKey.go:12]: Not really, will unexport it.

File: core/getNodeFromDBErrWithKey.go [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+import (
+	"encoding/hex"
+	"fmt"
+)
+
+// GetNodeFromDBErrorString represents the string which is returned when a getting node from DB returns an error
+const GetNodeFromDBErrorString = "getNodeFromDB error"
+
+// GetNodeFromDBErrWithKey defines a custom error for trie get node
```
Comment (iulianpascalau) [inline core/getNodeFromDBErrWithKey.go:11]: start comment with lower case to align with the function name?

**Review by iulianpascalau — DISMISSED** [review]

File: core/getNodeFromDBErrWithKey.go (hunk 8 above)
Comment (BeniaminDrasovean) [reply core/getNodeFromDBErrWithKey.go]: Changed.

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**